Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,212 @@
+    - MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+
+  - if the funding transaction confirms:
+    - MUST track spend of the funding output
+    - MUST advances its off-chain state to the execution phase
+    - MUST note its change output for later spend
+
+A node:
+
+  - if the funding input double-spend confirms:
+    - MUST abort the DLC and prevent further interactions with counterparty
+
+## Rationale
+
+Funding transaction must be confirmed to consider the off-chain state initialized. As mempools
+congestion may fluctuate between fees negotiation and transaction propagation, party must adjust
+unilaterally the package feerate (CPFP) according to their confirmation preferences. A party may 
+choose to wait for mempools draining before to adjust the feerate to avoid a fee waste as block 
+height of transition to execution phase doesn't matter, assuming it's before `contract_maturity_bound`.
+
+Reaching near `contract_maturiy_bound` requires to confirm the funding transaction quickly as it's
+now a security matter. After oracle signature release, a malicious party could try to double-spend
+an unconfirmed funding transaction to avoid losing its collateral, and thus without assuming deceitful
+cooperation with the oracle.
+
+
+# Execution Phase
+
+The execution phase transitions to the terminal phase either by a CET or a refund
+transaction confirmations.
+
+## Requirements
+
+A node:
+
+  - if an oracle signature is received:
+    - MUST finalize the CET with a local `SIGHASH_ALL` signature
+    - MUST broadcast the CET
+    - MAY spend its outcome output with a CPFP increasing package feerate to shorten confirmation time
+    - if the CET is not confirmed and `contract_timeout` is 20 blocks ahead of local tip:
+      - MUST rebroadcast the CET
+      - MUST spend its outcome output with a CPFP increasing package feerate to shorten confirmation time
+      - MUST start a fee-bumping timer of length `security_delay`
+    - if `security_delay` expires:
+      - MUST spend the change output with a replacing CPFP increasing package feerate to shorten confirmation time
+      - MUST replace the CPFP transaction according to BIP 125
+      - MUST reschedule timer for next bump
+    - MUST NOT spend more in bumping fees than its outcome value
+
+A node:
+
+  - if a CET transaction confirms:
+    - MUST note its outcome output for later spend
+    - MUST advances its off-chain state to terminal phase
+
+A node:
+
+  - if no oracle signature has been received and `contract_timeout` is reached:
+    - MUST finalize the refund transaction with a local `SIGHASH_ALL` signature
+    - MUST broadcast the refund transaction
+    - if the refund is not confirmed after `congestion_delay`:
+      - MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+      - MUST start a fee-bumping timer of length `congestion_delay`
+    - if `congestion_delay` expires:
```

Comment:
I'm not in favor of re-using the `congestion_delay` that is used for the funding transaction, I think that this should be explicitly called `refund_congestion_delay` as it will likely be a much larger number than the funding's `congestion_delay` because in the funding case you are worried about information around odds of outcomes changing incentivizing counter-party to double-spend inputs, whereas here in the refund case the only worry is that the oracle could be or become compromised which is likely less probable than information changing.

Reply:
See my reply on `contract_maturity_bound` first. I think a congestion fee-bumping strategy must be different from a security fee-bumping one to prevent mempool spikes provoking massive and useless fees burning for DLC currently in funding phase. I might be to much cautious here, but that's something we have to care about. At least, until we move off-chain where funding and its congestion management isn't blurred with DLC setting up (caveat with zero-conf channels).

So minding `congestion_delay` isn't for odds of outcomes incentivizing counter-party to double-spend inputs, but still conceding you the oracle compromise risk for refund is a different risk, we may a) introduce  a `refund_security_delay` to specifically address the risk of oracle compromise or b) replace directly `congestion_delay` by a security one to bound implementation complexity.